src/store/modules/products.ts

```diff
@@ -0,0 +1,59 @@
+import {ProductsState} from "@/store/interfaces";
+import {FoodListItem} from "@/store/navigationInterface";
+import {getFoodList} from "@/utilities/apiCalls";
+
+const state: ProductsState = {
+    products: [],
+    pageNumber: 1,
+    numberOfPages: 0,
+}
+
+const getters = {
+    getProducts(state: ProductsState) {
+        return state.products;
+    },
+    getPage(state: ProductsState) {
+        return state.pageNumber;
+    }
+}
+
+const mutations = {
+    nextPage(state: ProductsState) {
+        if (state.pageNumber < state.numberOfPages) {
+            state.pageNumber += 1;
+        }
+    },
+    prevPage(state: ProductsState) {
+        if (state.pageNumber > 1) {
+            state.pageNumber -= 1
+        }
+    },
+    setPage(state: ProductsState,page: number) {
+        state.pageNumber = 0;
+        if(page <= state.numberOfPages && page > 0) {
+            state.pageNumber = page;
+        } else state.pageNumber = 1;
+    },
+    setResults(state: ProductsState, list: Array<FoodListItem>) {
+        state.products = [];
+        for (let i = (state.pageNumber * 9 - 9); i < (state.pageNumber * 9); i++) {
+            if(list[i] !== undefined) {
+                state.products.push(list[i]);
+            }
+        }
+    },
+    setNumberOfPages(state: ProductsState, list: Array<FoodListItem>) {
+        if (list.length % 9 !== 0) {
```

Comment:
I see bare `9` all over the file. I assume that this is max number of items per page. What if suddenly client will want 12 items per page? Put all `9` in one variable and store it along with other global variables